### DISCUSSION TITLE
SQL schema support could be very useful

### DISCUSSION THREAD

**Discussion by hobby-AI** [SecureAI-Tools/SecureAI-Tools#78] [Ideas]
I tried Mistral model and it generated quite good SQL query on my request. If you can add .sql schema file support and allow model to use schema to generate good SQL query, it could become more useful?

**Comment by JayNakrani** [comment]
Oh, interesting. We can certainly add support for .SQL files. However, asking it to generate SQL query based on DB schema may not be a good use case for conversational question answering because it currently splits the document into smaller chunks. To generate SQL query from DB schema, we'd need to pass entire DB schema through LLM instead of splitting into smaller chunks. Let me think through how to solve this.

**Comment by JayNakrani** [comment]
@ConstructionQA In the meantime, you can create a chat without selecting any documents and include DB schema in your prompt. That should work.
